packages/react-components/src/components/Button/NavButton.jsx

```diff
@@ -0,0 +1,48 @@
+import { forwardRef } from 'react'
+import { func, bool, string, oneOf } from 'prop-types'
+import BaseButton from './BaseButton'
+import theme from '../theme'
+
+const NavButton = forwardRef(
+  ({ disabled, onClick, variant, title, type, isActive, ...props }, ref) => {
+    const activeColor = theme.colors.core.primary
+    const defaultColor = theme.colors.buttons.secondary.borderColor
+
+    // overrides for the base button
+    const buttonDefaultStyles = {
+      borderRadius: 50,
+      color: isActive ? activeColor : defaultColor,
+      borderColor: isActive ? activeColor : defaultColor,
+      borderWidth: isActive ? 2 : 'auto'
+    }
```

Comment:
It's not _quite_ as readable, but can we just define these things inline in the `BaseButton` so we can save ourselves declaring constants in the function? I'm pedantic with saving memory wherever possible, and I like to avoid defining new constants on each rerender

Reply:
Yeah sure, although those variable declarations should get removed anyway with the minifier right?